indico/modules/users/controllers.py

```diff
@@ -102,37 +123,56 @@ class RHUserDashboard(RHUserBase):
 
     def _process(self):
         self.user.settings.set('suggest_categories', True)
-        tz = session.tzinfo
-        hours, minutes = timedelta_split(tz.utcoffset(datetime.now()))[:2]
         categories = get_related_categories(self.user)
         categories_events = []
         if categories:
             category_ids = {c['categ'].id for c in categories.itervalues()}
-            today = now_utc(False).astimezone(tz).date()
-            query = (Event.query
-                     .filter(~Event.is_deleted,
-                             Event.category_chain_overlaps(category_ids),
-                             Event.start_dt.astimezone(session.tzinfo) >= today)
-                     .options(joinedload('category').load_only('id', 'title'),
-                              joinedload('series'),
-                              subqueryload('acl_entries'),
-                              load_only('id', 'category_id', 'start_dt', 'end_dt', 'title', 'access_key',
-                                        'protection_mode', 'series_id', 'series_pos', 'series_count'))
-                     .order_by(Event.start_dt, Event.id))
-            categories_events = get_n_matching(query, 10, lambda x: x.can_access(self.user))
+            categories_events = get_events_in_categories(category_ids, self.user)
         from_dt = now_utc(False) - relativedelta(weeks=1, hour=0, minute=0, second=0)
         linked_events = [(event, {'management': bool(roles & self.management_roles),
                                   'reviewing': bool(roles & self.reviewer_roles),
                                   'attendance': bool(roles & self.attendance_roles)})
                          for event, roles in get_linked_events(self.user, from_dt, 10).iteritems()]
         return WPUser.render_template('dashboard.html', 'dashboard',
-                                      offset='{:+03d}:{:02d}'.format(hours, minutes), user=self.user,
+                                      user=self.user,
                                       categories=categories,
                                       categories_events=categories_events,
                                       suggested_categories=get_suggested_categories(self.user),
                                       linked_events=linked_events)
 
 
+class RHExportDashboardICS(RHUserBase):
+    @use_kwargs({
+        'from_': HumanizedDate(data_key='from', missing=now_utc(False) - relativedelta(weeks=1)),
+        'include': fields.List(fields.Str(), missing={'linked', 'categories'}),
+        'limit': fields.Integer(missing=100, validate=lambda v: v > 0 and v <= 500)
```

Comment:
`0 < v <= 500`

Reply:
Always forget the Python can do chained inequalities.